staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go

```diff
@@ -341,7 +341,7 @@ func NewGenericServerResponse(code int, verb string, qualifiedResource schema.Gr
 		message = "the server reported a conflict"
 	case http.StatusNotFound:
 		reason = metav1.StatusReasonNotFound
-		message = "the server could not find the requested resource"
+		message = fmt.Sprintf("the server could not find the requested resource %v", qualifiedResource)
```

Comment:
these are translating generic server errors... I don't think it's correct to include the qualified resource here

Reply:
> these are translating generic server errors... I don't think it's correct to include the qualified resource here

yeah, I don't think we should add it here. Moreover, we have also got this:

https://github.com/kubernetes/kubernetes/blob/16c5ba4114bf6443c2ce8c0104ad520dbcece6ce/staging/src/k8s.io/apimachinery/pkg/api/errors/errors.go#L385-L389

which gives an error like `Error from server (NotFound): the server could not find the requested resource (get replicationcontrollers foo)`

The original issue was this:

> E.g. try to list a custom resource, that is not deployed. All you get is "the server could not find the requested resource"

I think it's coming from:

https://github.com/kubernetes/kubernetes/blob/16c5ba4114bf6443c2ce8c0104ad520dbcece6ce/staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go#L231

We could edit this to show the resource name as well.

/cc @sttts

Reply:
Thanks for all of your review,I've changed it like nikhita said and the resource just use requestInfo.Resource + "." + requestInfo.APIGroup.
WDYT?